app/src/main/java/uk/gov/onelogin/wallet/DeleteWalletDataUseCase.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.onelogin.wallet
+
+import androidx.fragment.app.FragmentActivity
+import javax.inject.Inject
+import uk.gov.android.wallet.sdk.WalletSdk
+
+fun interface DeleteWalletDataUseCase {
+    suspend operator fun invoke(activityFragment: FragmentActivity)
+}
+
+@Suppress(
+    "TooGenericExceptionCaught",
+    "RethrowCaughtException",
+    "TooGenericExceptionThrown"
+)
+class DeleteWalletDataUseCaseImpl @Inject constructor(
+    private val walletSdk: WalletSdk
+) : DeleteWalletDataUseCase {
+    override suspend operator fun invoke(activityFragment: FragmentActivity) {
+        try {
+            val deleteResult = walletSdk.deleteWalletData(activityFragment)
+            if (!deleteResult) throw Exception(DELETE_WALLET_DATA_ERROR)
```

Comment:
Please create a specific Exception to be thrown here

Reply:
Also, the big red warnings you've suppressed are there to help you. They're not there to get in your way, but to guide you to a cleaner / better solution that will help you in the long run. If you are suppressing warnings, then I would like a written dev comment detailing why, please 

P.S. expect the same from others, if you see us just suppress things, please call it out.